packages/react-reconciler/src/ReactFiberThrow.new.js

```diff
@@ -453,6 +453,10 @@ function throwException(
     const wakeable: Wakeable = (value: any);
     resetSuspendedComponent(sourceFiber, rootRenderLanes);
 
+    if (getIsHydrating() && sourceFiber.mode & ConcurrentMode) {
```

Comment:
This whole condition should be in DEV, I think.

Reply:
That would work today because didSuspend currently only affects warning behaviors in dev mode but this file doesn't know that and that could change in the future.

Is it apparent enough that we are currently only calling a function suffixed with DEV to remind us this block is dev only? I supposed if we renamed the function because it was now going to be useful for production behavior we would see it being used in a dev only block and change that?

Reply:
>Is it apparent enough that we are currently only calling a function suffixed with DEV to remind us this block is dev only?

Yeah. You can also rename `didSuspend` to `didSuspendDEV` to make sure we don't introduce dependencies.

Reply:
yeah I like that, thanks